project/SparkBuild.scala

```diff
@@ -852,6 +852,7 @@ object Unidoc {
       .map(_.filterNot(_.getCanonicalPath.contains("org/apache/spark/sql/hive/test")))
       .map(_.filterNot(_.getCanonicalPath.contains("org/apache/spark/sql/catalog/v2/utils")))
       .map(_.filterNot(_.getCanonicalPath.contains("org/apache/hive")))
+      .map(_.filterNot(_.getCanonicalPath.contains("org/apache/spark/sql/v2/avro")))
```

Comment:
shall we just rename the package? all other file source is v2 is in `org.apache.spark.sql.execution.datasources.v2`

Reply:
it's weird that avro doesn't follow this rule.

Reply:
Then we will have to move `org.apache.spark.sql.avro.AvroFileFormat` to `org.apache.spark.sql.execution.datasources.avro.AvroFileFormat` as well, to make it consistent with ParquetFileFormat.

Originally the package `org.apache.spark.sql.avro` is created for consistence with the kafka-sql module, since both of them are external modules.

Reply:
I think the problem here is we have public APIs in the avro module, while we assume it should all be private like other external modules.

I think a safer way is to use whitelist and only generate doc for the classes we explicitly specified in avro module. Is it possible to do so?

Reply:
The `ignoreUndocumentedPackages` here is to ignore some packages. I think it will be complex if we introduce another step for adding some package as whitelist(allowlist)